src/classes/ADDR_OtherAcc_TEST.cls

```diff
@@ -0,0 +1,204 @@
+        insert listAcc;
+
+        // update the accounts' address
+        for (Account acc : listAcc) {
+            acc.BillingStreet = '123 45th';
+            acc.BillingCity = 'Seattle';
+        }
+        Test.startTest();
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update listAcc;
+        Test.stopTest();
+
+        // verify results
+        listAcc = [select Id, Name, BillingStreet, BillingCity, BillingState, BillingPostalCode, BillingCountry, 
+                   Current_Address__c from Account order by Account.Name];
+        system.assertEquals(cAcc, listAcc.size());
+        
+        list<Address__c> listAddr = [select Id, Default_Address__c, MailingStreet__c, MailingCity__c, 
+        Parent_Account__c, Latest_Start_Date__c, Latest_End_Date__c from Address__c 
+        order by Parent_Account__r.Name];
+        system.assertEquals(cAcc, listAddr.size());
+                
+        for (Integer i = 0; i < cAcc; i++) {
+            Account acc = listAcc[i];
+            Address__c addr = listAddr[i];
+            system.assertEquals(acc.BillingStreet, addr.MailingStreet__c);
+            system.assertEquals(acc.BillingCity, addr.MailingCity__c);
+            system.assertNotEquals(null, acc.BillingStreet);
+            system.assertNotEquals(null, acc.BillingCity);
+            system.assertEquals(true, addr.Default_Address__c);
+            system.assertNotEquals(null, acc.Current_Address__c);
+        }
+    }
+    
+    /*********************************************************************************************************
+    @description Update N accounts with address objects 
+    verify:
+        account addresses updated
+        contact addresses not updated
+    **********************************************************************************************************/            
+    @isTest
+    public static void newAddrForAccounts() {           
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c(Accounts_Addresses_Enabled__c = 
+                                                                      UTIL_Describe.getBizAccRecTypeID() + ';'));
+ 
+        // create accounts without addresses
+        list<Account> listAcc = UTIL_UnitTestData_TEST.CreateMultipleTestAccounts(3, UTIL_Describe.getBizAccRecTypeID());
+        insert listAcc;
+
+         // create addresses for the accounts
+        List<Address__c> addrs = UTIL_UnitTestData_TEST.initTestAddr(listAcc.size());
+        for (Integer i = 0; i < listAcc.size(); i++) {
+            addrs[i].Parent_Account__c = listAcc[i].Id;
+            addrs[i].Default_Address__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        
+        // verify results
+        listAcc = [select Id, Name, BillingStreet, BillingCity, BillingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. I suggest to introduce a `WHERE` clause or cap the max number of records returned adding a `LIMIT` clause to the query.